### PR TITLE
Halves `OpenDreamClient` warnings

### PR DESCRIPTION
**Pull request by ike709**
Takes `OpenDreamClient` from 40 to 20 warnings.

I confirmed that the changes to gliding and the context menu didn't break them.

### CODE REVIEW

File: OpenDreamClient/Interface/InterfaceElement.cs [hunk 1]
```diff
@@ -42,7 +42,9 @@
 
             MappingDataNode newNode = original.Merge(node);
 
-            ElementDescriptor = (ElementDescriptor)serializationManager.Read(ElementDescriptor.GetType(), newNode);
+            var descriptor = serializationManager.Read(ElementDescriptor.GetType(), newNode);
+            if(descriptor is null) throw new NullReferenceException(); // We're in a try/catch anyway, just play it safe
```
[the remark this answers is not in the document]
Comment (ike709) [reply OpenDreamClient/Interface/InterfaceElement.cs:46]: I think complying with this results in yuckier-looking code in this case so I'll leave it to wixoa to decide.

File: OpenDreamClient/Interface/DreamInterfaceManager.cs [hunk 2]
```diff
@@ -891,21 +891,22 @@ private void LoadInterface(InterfaceDescriptor descriptor) {
     }
 
     private void LoadDescriptor(ElementDescriptor descriptor) {
+        // Note: Id has a getter to ensure Id.Value is never null
```
Comment (wixoaGit) [inline OpenDreamClient/Interface/DreamInterfaceManager.cs:894]: I don't see why `Value` is marked nullable in the first place. The value isn't nullable in the constructor, and nothing ever gives it `null`.

File: OpenDreamClient/Interface/DMF/IDMFProperty.cs [hunk 3]
```diff
@@ -226,9 +226,39 @@ public bool Equals(string comparison) {
         DMFPropertyVec2 comparisonVec = new(comparison);
         return comparisonVec.X == X && comparisonVec.Y == Y;
     }
+
+    public bool Equals(DMFPropertyVec2 other)
```
Comment (wixoaGit) [inline OpenDreamClient/Interface/DMF/IDMFProperty.cs:230]: All these new methods need their braces moved to the same line.

File: OpenDreamClient/Interface/DMF/IDMFProperty.cs [hunk 4]
```diff
@@ -226,9 +226,39 @@ public bool Equals(string comparison) {
...
+
+    public override bool Equals(object? obj)
+    {
+        return obj is DMFPropertySize other && Equals(other);
+    }
+
+    public override int GetHashCode()
+    {
+        return _value.GetHashCode();
+    }
```
Comment (wixoaGit) [inline OpenDreamClient/Interface/DMF/IDMFProperty.cs:260]: Move these to the bottom

File: OpenDreamClient/Interface/DMF/IDMFProperty.cs [hunk 5]
```diff
@@ -299,7 +329,22 @@ public bool Equals(string comparison) {
...
+
+    public override bool Equals(object? obj)
+    {
+        return obj is DMFPropertyPos other && Equals(other);
+    }
+
+    public override int GetHashCode()
+    {
+        return _value.GetHashCode();
+    }
```
Comment (wixoaGit) [inline OpenDreamClient/Interface/DMF/IDMFProperty.cs:346]: Move these to the bottom

File: OpenDreamClient/Interface/InterfaceElement.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (wixoaGit) [reply OpenDreamClient/Interface/InterfaceElement.cs:46]: I'm fine with it as is.

**Review by wixoaGit — CHANGES_REQUESTED** [review]

**Review by wixoaGit — APPROVED** [review]

**Pull request merged**